src/applications/static-pages/homepage/HomepageSearch.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { VaSearchInput } from '@department-of-veterans-affairs/web-components/react-bindings';
+import PropTypes from 'prop-types';
+import * as Sentry from '@sentry/browser';
+
+import { replaceWithStagingDomain } from '../../../platform/utilities/environment/stagingDomains';
+import { apiRequest } from '../../../platform/utilities/api';
+
+/**
+ * Homepage redesign
+ * Search component that appears in the Common Tasks section (midpage)
+ * which uses the new Search Input component from the VA Design System
+ */
+const HomepageSearch = ({ value, suggestions }) => {
+  const [userInput, setUserInput] = useState(value);
+  const [latestSuggestions, setLatestSuggestions] = useState(suggestions);
+
+  // fetch Typeahead suggestions from API
+  const fetchDropDownSuggestions = async inputValue => {
```

Comment:
This function is doing two things that are likely redundant (and potentially buggy).

1. It returns a value that is used to set the state of the list of suggestions.
2. It sets that state directly, as well.

We don't need both of these to happen. The decision of how to fix it revolves around determining what the purpose of the function is.  Is its purpose to fetch some results and return them? Or is its purpose to fetch some results _and_ update state based on those results? I generally prefer the first (single-responsibility principle). If we go this route, we'll need to keep in mind that it is asynchronous, and we'll have to handle things accordingly when we call this function (see comment below about using a promise chain or await).